dash-pipeline/bmv2/stages/conntrack_lookup.p4

```diff
@@ -0,0 +1,220 @@
+        EthernetAddress dst_mac,
+        IPv4ORv6Address sip,
+        IPv4ORv6Address dip,
+        IPv6Address sip_mask,
+        IPv6Address dip_mask,
+      
+        /* Extra flow metadata */ 
+        @SaiVal[type="sai_u8_list_t"] bit<16> vendor_metadata,
+        @SaiVal[type="sai_u8_list_t"] bit<16> flow_data_pb)
+    {
+        /* Set Flow basic metadata */
+        meta.conntrack_data.flow_data.version = version;
+        meta.conntrack_data.flow_data.dash_direction = dash_direction;
+        meta.conntrack_data.flow_data.actions = dash_flow_action;
+        meta.meter_class = meter_class;
+        meta.conntrack_data.is_unidirectional_flow = is_unidirectional_flow;
+        
+        /* Set reverse flow key */
+        meta.conntrack_data.reverse_flow_key.eni_mac = reverse_flow_eni_mac;
+        meta.conntrack_data.reverse_flow_key.vnet_id = reverse_flow_vnet_id;
+        meta.conntrack_data.reverse_flow_key.ip_proto = reverse_flow_ip_proto;
+        meta.conntrack_data.reverse_flow_key.src_ip = reverse_flow_src_ip;
+        meta.conntrack_data.reverse_flow_key.dst_ip = reverse_flow_dst_ip;
+        meta.conntrack_data.reverse_flow_key.src_port = reverse_flow_src_port;
+        meta.conntrack_data.reverse_flow_key.dst_port = reverse_flow_dst_port;
+
+
+        /* Set encapsulation metadata */
+        meta.encap_data.vni = underlay0_vnet_id;
+        meta.encap_data.underlay_sip = underlay0_sip;
+        meta.encap_data.underlay_dip = underlay0_dip;
+        meta.encap_data.dash_encapsulation = underlay0_dash_encapsulation;
+
+        meta.tunnel_data.vni = underlay1_vnet_id;
+        meta.tunnel_data.underlay_sip = underlay1_sip;
+        meta.tunnel_data.underlay_dip = underlay1_dip;
+        meta.tunnel_data.dash_encapsulation = underlay1_dash_encapsulation;
+        meta.tunnel_data.underlay_smac = underlay1_smac;
+        meta.tunnel_data.underlay_dmac = underlay1_dmac;
+
+
+        /* Set overlay rewrite metadata */
+        meta.overlay_data.dmac = dst_mac;
+        meta.overlay_data.sip = sip;
+        meta.overlay_data.dip = dip;
+        meta.overlay_data.sip_mask = sip_mask;
+        meta.overlay_data.dip_mask = dip_mask;
+    }
+
+    @SaiTable[name = "flow", api = "dash_flow", order = 1, enable_bulk_get_api = "true", enable_bulk_get_server = "true"]
+    table flow_entry {
+        key = {
+            meta.conntrack_data.flow_key.eni_mac : exact;
+            meta.conntrack_data.flow_key.vnet_id : exact;
+            meta.conntrack_data.flow_key.ip_proto : exact;
+            meta.conntrack_data.flow_key.src_ip : exact;
+            meta.conntrack_data.flow_key.dst_ip : exact;
+            meta.conntrack_data.flow_key.src_port : exact;
+            meta.conntrack_data.flow_key.dst_port : exact;
+            meta.conntrack_data.flow_key.is_ipv6 : exact @SaiVal[name = "src_ip_is_v6"];
+        }
+
+        actions = {
+            set_flow_entry_attr;
+        }
+    }
+    
+    //
+    // Flow bulk get session filter:
```

Comment:
It will be great to add a comment here to indicate these tables are for api generation only and will not affects how data plane works.